### PR TITLE
Doc/extension guide

### PR DESCRIPTION
**Pull request by askulkarni2**
### What does this PR do?

Creates a new Extensibility guide for Customers and Partners to provide their add-ons via the framework.

### Motivation

We want to provide a way for customers and partners to be able easily extend the framework so add-ons can be quickly added outside of the core repo.

### More

- [X] Yes, I have tested the PR using my local account setup  (Provide any test evidence report under Additional Notes)
- [X] Yes, I have added a new example under [examples](https://github.com/aws-samples/aws-eks-accelerator-for-terraform/tree/main/examples) to support my PR
- [X] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/ssp-eks-add-ons) repo (if applicable)
- [X] Yes, I have updated the [docs](https://github.com/aws-samples/aws-eks-accelerator-for-terraform/tree/main/docs) for this feature
- [X] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs required examples and docs except a new pattern or add-on added.

### For Moderators
- [x] E2E Test successfully complete before merge?

### Additional Notes

See the demo add-on implementation at https://github.com/askulkarni2/terraform-eksblueprints-kube-state-metrics-addon.

### CODE REVIEW

File: docs/extensibility.md [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+- The blueprints code contains the code that demonstrates how customers can use your add-on with the EKS Blueprints framework. Here, we highly recommend that you show the true value add of your add-on through the pattern. Customers will benefit the most where the example shows how they can integrate their workload with your add-on.
+
+If your add-on can be deployed via helm chart, we recommend the use of the [helm-addon](../modules/kubernetes-addons/helm-addon) as shown below.
+
+> main.tf
+```hcl
+module "helm_addon" {
+  source            = "github.com/aws-samples/aws-eks-accelerator-for-terraform//modules/kubernetes-addons/helm-addon"
+  manage_via_gitops = var.manage_via_gitops
+  set_values        = local.set_values
```
Comment (vara-bonthu) [inline docs/extensibility.md:124]: show `set_sensitive_values` as well here.

File: docs/extensibility.md [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+  type        = bool
+  default     = false
+  description = "Enable Kube State Metrics add-on"
+}
+
+variable "kube_state_metrics_helm_config" {
+  type        = any
+  default     = {}
+  description = "Kube State Metrics Helm Chart config"
+}
```
Comment (vara-bonthu) [inline docs/extensibility.md:162]: We need the following as well

```hcl
variable "kube_state_metrics_irsa_permissions_boundary" {
  type        = string
  default     = ""
  description = "IAM Policy ARN for IRSA IAM role permissions boundary"
}

variable "kube_state_metrics_irsa_policies" {
  type        = list(string)
  default     = []
  description = "IAM policy ARNs for Kube State Metrics IRSA"
}

```

File: modules/kubernetes-addons/variables.tf [hunk 3]
```diff
@@ -441,3 +441,22 @@ variable "argo_rollouts_irsa_policies" {
...
+  default     = false
+  description = "Enable Kube State Metrics add-on"
+}
+
+variable "kube_state_metrics_helm_config" {
+  type        = any
+  default     = null
+  description = "Kube State Metrics Helm Chart config"
+}
+
```
Comment (vara-bonthu) [inline modules/kubernetes-addons/variables.tf:457]: Add this here

```hcl
variable "kube_state_metrics_irsa_permissions_boundary" {
  type        = string
  default     = ""
  description = "IAM Policy ARN for IRSA IAM role permissions boundary"
}
```

**Review by vara-bonthu — CHANGES_REQUESTED** [review]
Nice on Apoorva! 👍🏼  
Few minor changes requested

File: docs/extensibility.md [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+2. We recommend, you implement a separate repository for your private add-on while still using the upstream framework. This gives you the advantage of keeping up with ongoing feature releases and bug fixes while keeping your add-on private.
+
+### Secrets Handling
+
+We expect that certain add-ons will need to provide to sensitive values to their helm chart configuration such as password, license keys, API keys, etc. We recommend that you ask customers to store such secrets in an external secret store such as AWS Secrets Manager or AWS Systems Manager Parameter Store and use the [AWS Secrets and Configuration Provider (ASCP)](https://docs.aws.amazon.com/secretsmanager/latest/userguide/integrating_csi_driver.html) to mount the secrets as files or environment variables in the pods of your add-on. We are actively working on providing a native add-on for ASCP as of this writing which you will be able to levarage for your add-on.
```
Comment (kcoleman731) [inline docs/extensibility.md:54]: provide access to

**Review by kcoleman731 — COMMENTED** [review]
Really nice work here.

**Review by vara-bonthu — APPROVED** [review]
LGTM

**Review by kcoleman731 — COMMENTED** [review]
This looks awesome. Thank you!

**Pull request merged**